install/kubernetes/cilium/templates/cilium-agent-servicemonitor.yaml

```diff
@@ -1,11 +1,11 @@
-{{- if and .Values.global.prometheus.enabled (.Values.global.prometheus.serviceMonitor.enabled) }}
+{{- if and (.Values.agent) (not .Values.preflight.enabled) (.Values.prometheus.enabled) (.Values.prometheus.serviceMonitor.enabled) }}
```

Comment:
Will it fail if we don't have `monitoring.coreos.com/v1` pre-installed ?

We can either mention docs for the prerequisites, or change condition to something like `.Capabilities.APIVersions.Has "monitoring.coreos.com/v1"` or even do both ?

Keen to hear your input @errordeveloper @seanmwinn

Reply:
not necessary to do in this PR though.

Reply:
I think that we should indeed add `.Capabilities.APIVersions.Has "monitoring.coreos.com/v1"`, perhaps even drop `.Values.global.prometheus.serviceMonitor.enabled`.

Reply:
https://github.com/cilium/cilium/issues/13513